pkg/cloud/services/network/natgateways.go

```diff
@@ -287,7 +287,11 @@ func (s *Service) deleteNatGateway(id string) error {
 		}
 
 		if len(out.NatGateways) == 0 {
-			return false, errors.Wrapf(err, "no NAT gateway returned for id %q", id)
+			if err != nil {
+				return false, errors.Wrapf(err, "no NAT gateway returned for id %q", id)
+			}
+			errmsg := fmt.Sprintf("no NAT gateway returned for id %q", id)
+			return false, errors.New(errmsg)
```

Comment:
nit: We can directly use `fmt.Sprintf("no NAT gateway returned for id %q", id)` as an argument to `errors.New()` without introducing new variable, wdyt?

Reply:
yes done them that way